pkg/proc/evalop/evalcompile.go

```diff
@@ -112,9 +112,38 @@ func CompileSet(lookup evalLookup, lhexpr, rhexpr string) ([]Op, error) {
 }
 
 func (ctx *compileCtx) compileAllocLiteralString() {
-	ctx.pushOp(&CallInjectionAllocString{Phase: 0})
-	ctx.pushOp(&CallInjectionAllocString{Phase: 1})
-	ctx.pushOp(&CallInjectionAllocString{Phase: 2})
+	jmp := &Jump{When: JumpIfAllocStringChecksFail}
+	ctx.pushOp(jmp)
+
+	ctx.compileSpecialCall("runtime.mallocgc", []ast.Expr{
+		&ast.BasicLit{Kind: token.INT, Value: "0"},
+		&ast.Ident{Name: "nil"},
+		&ast.Ident{Name: "false"},
+	}, []Op{
+		&PushLen{},
+		&PushNil{},
+		&PushConst{constant.MakeBool(false)},
+	})
+
+	ctx.pushOp(&ConvertAllocToString{})
+	jmp.Target = len(ctx.ops)
```

Comment:
Because the target of the jump, the next instruction executed if the jump is taken, is `len(ctx.ops)`. The reason `evalJump` does `-1` is because `executeOp` always increments `opidx` by one after evaluating one instruction at https://github.com/go-delve/delve/blob/2e88b7ead3e2922f0853cd4dd934f85f9f7e8fd2/pkg/proc/eval.go#L1127.

Setting Target like this is easier to understand when we print instruction listings.